### ISSUE TITLE
🐛 BUG: glob support and Astro.fetchContent()

### ISSUE THREAD

**Issue by brycewray** [snowpackjs/astro#1098]
### What package manager are you using?

npm

### What operating system are you using?

macOS 11.5.2

### Describe the Bug

**Note**: This is observed in Astro 0.19.0-next.2.

Given the existence of `src/pages/posts/2021/07/test.md` along with multiple `.md` files at the `src/pages/posts/` level, the following correctly accesses the latter set of files:

```astro
---
const posts = Astro.fetchContent('posts/*.md')
---
```

...but trying a glob to access all the `.md` files, including the nested ones, within `src/pages/posts/`...

```astro
---
const posts = Astro.fetchContent('posts/**/*.md')
---
```
...causes the dev server to lock up on `npm run start` and builds to hang without completion on `npm run build`.

### Steps to Reproduce

**Note**: the following presumes the use of aliasing in `snowpack.config.mjs` so that `posts` means `src/pages/posts`.

1. `npm init astro` using any template.
2. Create a layout or component with this in the frontmatter script: `const posts = Astro.fetchContent('posts/*.md')`.
3. In the body of the layout or component, add the following: 
```astro
{posts.sort((a, b) => a.date.localeCompare(b.date))).map(post => (
  <div>
   <p><a href={post.url}>{post.title}</a>
  </div>
))}
```
4. Create an Astro file (`.astro`) which uses this layout or component.
5. Create one or more `.md` files within `src/pages/posts/` and at least one `.md` file within a more deeply nested folder in `src/pages/posts/` (e.g., `src/pages/posts/2021/07/test.md`). The frontmatter of each `.md` file should have a `title` key and entry.
6. `npm run start` and access the file using the layout or component.
7. Observe that the resulting output from the file using the layout or component ignores the nested file but lists the others.
8. Then go back to the layout or component and change the frontmatter script from step 2 to: `const posts = Astro.fetchContent('/posts/**/*.md')`.
9. `npm run start` and access the file using the layout of component.
10. This will result in the dev server hanging.
11. Similarly, `npm run build` results in a frozen build process.

### Link to Minimal Reproducible Example (Optional)

https://github.com/brycewray/astro-site

**Comment by brycewray** [comment]
Still seeing this behavior. Any thoughts?

**Comment by brycewray** [comment]
@drwpow  is there any news on this front? I'm stuck without a fix for this one.

**Comment by FredKSchott** [comment]
I don't think that anyone has been able to debug this, but if there's an issue here it is probably in Snowpack's `import.meta.glob` support, since we use that under the hood.
